bundles/binding/org.openhab.binding.weather/src/main/java/org/openhab/binding/weather/internal/bus/WeatherBinding.java

```diff
@@ -71,15 +71,22 @@ public void deactivate() {
      */
     @Override
     public void updated(Dictionary<String, ?> config) throws ConfigurationException {
-        if (config != null) {
-            context.getJobScheduler().stop();
+        if (config == null) {
+            logger.warn("Unable to find any configuration settings for weather binding. Check openhab.cfg.");
+            throw new ConfigurationException("weather", "Unable to find any configuration settings for weather binding. Check openhab.cfg.");
+        }
```

Comment:
You're thinking we shouldn't throw an exception here, but just issue a warning? I'd be ok with that.

I can't think of any situations when <code>updated</code> would be called with a <code>null</code> config, though. I've run through quite a number of test scenarios without seeing any bad behaviour.

At any rate, if you prefer to not throw an exception here, I'll change it.